### PR TITLE
#7645 correct missing constraints on queue name and acceptable charac…

### PR DESCRIPTION
**Pull request by rjmcginness**
fix: correct missing constraints on queue name and acceptable characters

#7645

### CODE REVIEW

File: tests/integration/test_sqs.py [hunk 1]
```diff
@@ -136,6 +137,20 @@ def test_create_queue_and_get_attributes(self, sqs_client, sqs_queue):
         assert int(float(attrs["CreatedTimestamp"])) == pytest.approx(int(time.time()), 30)
         assert int(attrs["VisibilityTimeout"]) == 30, "visibility timeout is not the default value"
 
+    @pytest.mark.only_localstack
+    def test_queue_name_too_long(self, sqs_create_queue, snapshot):
+        with pytest.raises(InvalidParameterValue) as e:
+            sqs_create_queue(QueueName=(81 * "a"))
+
+        snapshot.match("queue_name_invalid_character", e.value.response)
```
Comment (thrau) [inline tests/integration/test_sqs.py:145]: this seems like something we could test with snapshotting. here are a couple of examples. if you don't have an AWS account you want to test it with, then i'm happy to help with tests. https://github.com/localstack/localstack/blob/27b4e328cd2fbc3c7d3a11ee5d88c771da641d08/tests/integration/test_sqs.py#L1408-L1422

File: localstack/services/sqs/provider.py [hunk 2]
```diff
@@ -669,6 +671,12 @@ def create_queue(
 
         return CreateQueueResult(QueueUrl=queue.url(context))
 
+    def _verify_valid_queue_name(self, queue_name: str):
+        name_re = re.compile(sqs_constants.QUEUE_NAME_REGEX + "{" + f"{len(queue_name)}" + "}")
+        if not queue_name or len(queue_name) > 80 or name_re.match(queue_name) is None:
+            error_msg = "Can only include alphanumeric characters, hyphens, or underscores. 1 to 80 in length"
+            raise InvalidParameterValue(error_msg)
```
Comment (thrau) [inline localstack/services/sqs/provider.py:678]: we actually already have a method for this: https://github.com/localstack/localstack/blob/44ff8c78f3864c694dc52a44e0bfad032105be0f/localstack/services/sqs/provider.py#L105-L119

**Review by thrau — CHANGES_REQUESTED** [review]
thanks for the contribution @rjmcginness 

this one's a headscratcher to me because we already have a method for asserting the queue name, but it's apparently not in use. we do also have assert methods in the sqs queue model itself, which *are* used, so i'm not sure why these exceptions wouldn't show up https://github.com/localstack/localstack/blob/27b4e328cd2fbc3c7d3a11ee5d88c771da641d08/localstack/services/sqs/models.py#L462-L466

verifying the queue name directly in `create_queue` makes sense to me, but we could maybe re-use the method we already have (linked in it a comment). and then remove the validation from the `SqsQueue` class accordingly. i just want to avoid having the same code in three different ways.

and it would be great to add snapshot validated tests. if that's too much of a bother for you, i'm happy to help.

**Comment by rjmcginness** [comment]
@thrau Thank you, Thomas. Yes, makes perfect sense. I can try to do this, though it might not be quick. I will try to investigate why the other method is not working and use that in the place I added the new one.

**Pull request closed without merging**